pkg/apis/build/v1alpha1/build_template_validation.go

```diff
@@ -32,38 +24,8 @@ func (b *BuildTemplateSpec) Validate() *apis.FieldError {
 	return nil
 }
 
-func validateObjectMetadata(meta metav1.Object) *apis.FieldError {
-	name := meta.GetName()
-
-	if strings.Contains(name, ".") {
-		return &apis.FieldError{
-			Message: "Invalid resource name: special character . must not be present",
-			Paths:   []string{"name"},
-		}
-	}
-
-	if len(name) > maxLength {
-		return &apis.FieldError{
-			Message: "Invalid resource name: length must be no more than 63 characters",
-			Paths:   []string{"name"},
-		}
-	}
-	return nil
-}
-
-func validateParameters(params []ParameterSpec) *apis.FieldError {
-	// Template must not duplicate parameter names.
-	seen := map[string]struct{}{}
-	for _, p := range params {
-		if _, ok := seen[p.Name]; ok {
-			return apis.ErrInvalidKeyName("ParamName", "b.spec.params")
-		}
-		seen[p.Name] = struct{}{}
-	}
-	return nil
-}
-
-func validateVolumes(volumes []corev1.Volume) *apis.FieldError {
+//ValidateVolumes for any build CRD
+func ValidateVolumes(volumes []corev1.Volume) *apis.FieldError {
```

Comment:
Nice, glad we can share this. 👍 

The comment seems incomplete though, and could use a ` ` after `//`